packages/textsubsuper/init.lua

```diff
@@ -0,0 +1,374 @@
+end
+
+function package:_init ()
+  base._init(self)
+  self.class:loadPackage("inputfilter")
+end
+
+function package.declareSettings (_)
+  SILE.settings:declare({
+    parameter = "textsubsuper.scale",
+    type = "integer",
+    default = 0.66,
+    help = "Size scaling ratio of a fake superscript or subsscript"
+  })
+
+  SILE.settings:declare({
+    parameter = "textsubsuper.bolder",
+    type = "integer",
+    default = 200,
+    help = "Weight increase of a fake superscript or subsscript (e.g. 200 for normal to semibold)"
+  })
+
+  SILE.settings:declare({
+    parameter = "textsubsuper.vscale.number",
+    type = "number",
+    default = 0.90,
+    help = "Vertical ratio applied to numbers in fake superscript or subsscript"
+  })
+
+  SILE.settings:declare({
+    parameter = "textsubsuper.vscale.other",
+    type = "number",
+    default = 0.95,
+    help = "Vertical ratio applied to numbers in fake superscript or subsscript"
+  })
+
+  SILE.settings:declare({
+    parameter = "textsubsuper.offset.superscript",
+    type = "measurement",
+    default = SILE.measurement("0.70ex"),
+    help = "Offset of a fake superscript above the baseline (logically in a font-relative unit such as ex)"
+  })
+
+  SILE.settings:declare({
+    parameter = "textsubsuper.offset.subscript",
+    type = "measurement",
+    default = SILE.measurement("0.25ex"),
+    help = "Offset of a fake subscript below the baseline (logically in a font-relative unit such as ex)"
+  })
+end
+
+function package:registerCommands ()
+  local function rescaleFilter (node, content, args)
+    if type(node) == "table" then return node end
+    local result = {}
+    local chars = SU.splitUtf8(node)
+    for _, char in ipairs(chars) do
+      if not tonumber(char) then
+        result[#result+1] = self.class.packages.inputfilter:createCommand(
+          content.pos, content.col, content.line,
+          "textsubsuper:scale", {
+            xRatio = args.xScale,
+            yRatio = args.yScaleOther
+          }, { char }
+        )
+      else
+        result[#result+1] = self.class.packages.inputfilter:createCommand(
+          content.pos, content.col, content.line,
+          "textsubsuper:scale", {
+            xRatio = args.xScale,
+            yRatio = args.yScaleNumber
+          }, { char }
+        )
+      end
+    end
+    return result
+  end
+
+  local function rescaleContent(content)
+    local transformed
+    if SILE.outputter._name ~= "libtexpdf" then
+      -- Not supported
+      transformed = content
+    else
+      transformed = self.class.packages.inputfilter:transformContent(content, rescaleFilter, {
+        xScale = 1,
+        yScaleNumber = SILE.settings:get("textsubsuper.vscale.number"),
+        yScaleOther = SILE.settings:get("textsubsuper.vscale.other"),
+      })
+    end
+    SILE.process(transformed)
+   end
+
+  -- REAL SUPERSCRIPT / SUBSCRIPT WHEN AVAILABLE
+
+  self:registerCommand("textsuperscript", function (options, content)
```

Comment:
Why prefix these commands with "text". Would `\superscript` be okay? I can potentially see more than one package doing these things in a different way, but I would also expect them to be mutually exclusive. No?

Reply:
I think due to the maths package it's a no

Reply:
Math mode has a completely different parser and command space. No regular SILE commands are available in the math language and vice versa.

Reply:
Fair enough. I'm not sure whether it is the TeX feel or sheer verbosity that bothers me, but `sub`/`sup` doesn't sound bad. Opinion on that @Omikhleia?

(and pinging @OlivierNicole for kudos on there *being* a math mode)

Reply:
I'd advocate that there are _good_ reasons for these commands to be called `\textsub/super` rather than `\sub/super`, IMHO.
- They are not _general_ sub/super (à la HTML) that would work on any type of content (e.g. boxes, etc. - for that, we already have "raiselower"...).
- ... but rather, they work on "text" strings:
  - Trying to use font features when available...
  - Trying to redimension/scale some things e.g. digits...
  - Trying to use semibold (following Bringhurst) when they can etc...
- I.e., overall, trying to get a "decent typographic" output for e.g. 12th century, issue n° 2 (without a degree), Mlle ("mademoiselle"), etc. - That's again text sub/superscripted text bits, but not general super/subscript (you can't even expect them to nest with several levels...)

N.B. 
>  (It's really impressive, @Omikhleia et al. did a great job.) 
I had not involvement in math.... But the math stuff is unrelated, here we are dealing with pure text things, i.e. typography for raised/lowered text outside the context of math typesetting (again: centuries, abbreviations, numbers, etc. - topics that don't mandate math to even be supported and have other, albeit similar, constraints).

Reply:
BTW: Who cares about (La)TeX possibly similar names? There are certainly dozens of existing TeX/LaTeX packages that use similar names (_confer_, I even discovered an "enumitem" package I had no prior knowledge of, and anyway we ended up calling this "lists" - Wait, I am sure some LaTeX stuff already used those names for something else...). Sometimes we just have to stop running after TeX, no? (Esp. with regards to math, when we claim compliance to MathML over TeX, so why bother?)